### PR TITLE
Fix for issue #1800. 

### PR DESCRIPTION
**Pull request by schandrika**
# Description

Fix for issue #1800. Since start-volttron can now call start of rabbitmq server the start volttron process takes a while to complete all the 'on start' operation. So command like vctl status can fail in the mean time. 

Fixes # (issue)

main.py now created a pid file once volttron is done with all the on start operation. The same is deleted at shutdown. vctl and start-volttron scripts now check for this file and provide appropriate message to user. 
 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Tested both manually and added pytest

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/VOLTTRON/volttron/pull/1881%23discussion_r249564545%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1881%23discussion_r251545837%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1881%23discussion_r251545956%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%204f5e8fc07457337e466cd03087acff8eb9808f72%20volttrontesting/testutils/test_platformwrapper2.py%2021%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1881%23discussion_r251545837%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22removed%22%2C%20%22created_at%22%3A%20%222019-01-28T18%3A51%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20volttrontesting/testutils/test_platformwrapper2.py%3AL39-101%22%7D%2C%20%22Pull%204f5e8fc07457337e466cd03087acff8eb9808f72%20volttron/platform/control.py%2021%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1881%23discussion_r249564545%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Isn%27t%20this%20todo%20what%20you%20add%20with%20this%20pull%20request%3F%22%2C%20%22created_at%22%3A%20%222019-01-21T19%3A53%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/3979063%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/craig8%22%7D%7D%2C%20%7B%22body%22%3A%20%22Removed%20the%20second%20TODO.%20%22%2C%20%22created_at%22%3A%20%222019-01-28T18%3A51%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20volttron/platform/control.py%3AL2495-2516%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 4f5e8fc07457337e466cd03087acff8eb9808f72 volttron/platform/control.py 21'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1881#discussion_r249564545'>File: volttron/platform/control.py:L2495-2516</a></b>
- <a href='https://github.com/craig8'></a> Isn't this todo what you add with this pull request?
- <a href='https://github.com/schandrika'></a> Removed the second TODO.
- [ ] <a href='#crh-comment-Pull 4f5e8fc07457337e466cd03087acff8eb9808f72 volttrontesting/testutils/test_platformwrapper2.py 21'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1881#discussion_r251545837'>File: volttrontesting/testutils/test_platformwrapper2.py:L39-101</a></b>
- <a href='https://github.com/schandrika'></a> removed

<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1881?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1881?approve=1'></a>&nbsp;<a href='https://github.com/VOLTTRON/volttron/pull/1881'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: volttron/platform/control.py [hunk 1]
```diff
@@ -2493,6 +2495,21 @@ def add_parser(*args, **kwargs):
...
     args = argv[1:]
+
+    # TODO: for auth some of the commands will work when volttron is down and
+    # some will error (example vctl auth serverkey). Do check inside auth
+    # function
+    # Below vctl commands can work even when volttron is not up. For others
+    # volttron need to be up.
+    if args[0] not in ('list', 'tag', 'auth', 'rabbitmq'):
+        # check pid file
+        # TODO: is there a platform level constant file to save 'VOLTTRON_PID'
```
Comment (craig8) [inline volttron/platform/control.py]: Isn't this todo what you add with this pull request?

File: volttrontesting/testutils/test_platformwrapper2.py [hunk 2]
```diff
@@ -35,3 +39,62 @@ def test_can_cleanup_installed_listener():
     # give operating system enough time to update pids.
     gevent.sleep(0.1)
     assert not psutil.pid_exists(started)
+
+@pytest.mark.dev
```
Comment (craig8) [inline volttrontesting/testutils/test_platformwrapper2.py]: remove dev marker

Comment (schandrika) [reply volttrontesting/testutils/test_platformwrapper2.py]: removed

File: volttron/platform/control.py (hunk 1 above)
Comment (schandrika) [reply volttron/platform/control.py]: Removed the second TODO.

**Pull request merged**